runtime/dancebox/src/migrations.rs

```diff
@@ -209,19 +210,56 @@ where
     }
 }
 
+pub struct PolkadotXcmMigration<T>(pub PhantomData<T>);
+impl<T> Migration for PolkadotXcmMigration<T>
+where
+    T: pallet_xcm::Config,
+{
+    fn friendly_name(&self) -> &str {
+        "MM_PolkadotXcmMigration"
+    }
+
+    fn migrate(&self, _available_weight: Weight) -> Weight {
+        pallet_xcm::migration::v1::VersionUncheckedMigrateToV1::<T>::on_runtime_upgrade()
+    }
+}
+
+pub struct XcmpQueueMigration<T>(pub PhantomData<T>);
+impl<T> Migration for XcmpQueueMigration<T>
+where
+    T: cumulus_pallet_xcmp_queue::Config,
+{
+    fn friendly_name(&self) -> &str {
+        "MM_XcmpQueueMigration"
+    }
+
+    fn migrate(&self, _available_weight: Weight) -> Weight {
+        cumulus_pallet_xcmp_queue::migration::Migration::<T>::on_runtime_upgrade()
```

Comment:
do we know for certain this puts the v3 on the pallet?

Reply:
because we probably need to set v2 first before doing this on_runtime_upgrade

Reply:
Yes, even if onchain is v1, it applies first the migration to v2 and immediately after the migration to v3 https://github.com/paritytech/polkadot-sdk/blob/master/cumulus/pallets/xcmp-queue/src/migration.rs#L36C1-L46C4